### PR TITLE
Rely on org.sonarqube code for setting parameters

### PR DESCRIPTION
**Pull request by apetres**
It already implements the extension and the way of computing parameters,
including a set of default values. No need to implement our own logic.
Also this way we maintain compatibility with org.sonarqube plugin.

### CODE REVIEW

File: src/main/groovy/com/ullink/gradle/sonar/SonarScannerMsbuildPlugin.groovy [hunk 1]
```diff
@@ -35,15 +45,15 @@ class SonarScannerMsbuildPlugin implements Plugin<Project> {
...
             }
         }
     }
 
-    private void initSonarScanner(Project project, SonarScannerMsbuildExtension extension) {
+    private void initSonarScanner(Project project) {
         downloadSonarScanner(project)
         project.exec {
-            commandLine = [getSonarScannerFile(project), 'begin'] + buildArgs(project, extension.properties.properties)
+            commandLine = [getSonarScannerFile(project), 'begin'] + buildArgs(computeSonarProperties(project))
```
Comment (tephe) [inline src/main/groovy/com/ullink/gradle/sonar/SonarScannerMsbuildPlugin.groovy:56]: Do you know if sonarqube supports the help command? /?  or /h

File: src/main/groovy/com/ullink/gradle/sonar/SonarScannerMsbuildPlugin.groovy [hunk 2]
```diff
@@ -18,12 +21,19 @@ class SonarScannerMsbuildPlugin implements Plugin<Project> {
     static final SONAR_SCANNER_EXE = 'SonarScanner.MSBuild.exe'
     // keep the 'sonarqube' name for compatibility with the org.sonarqube plugin
     static final SONAR_SCANNER_TASK_NAME = 'sonarqube'
-    static final PROJECT_KEY_ARG = 'sonar.projectKey'
-    static final MANDATORY_ARGS = [n: 'sonar.projectName', v: 'sonar.projectVersion']
+    static final MANDATORY_ARGS = [k: 'sonar.projectKey',
+                                   n: 'sonar.projectName',
+                                   v: 'sonar.projectVersion'
+    ]
+    static final IGNORED_PROPERTIES = [
```
Comment (tephe) [inline src/main/groovy/com/ullink/gradle/sonar/SonarScannerMsbuildPlugin.groovy:28]: If we don't ignore it then SonarScanner.MSbuild.exe will throw an exception? Won't it be ignored anyway?

File: src/main/groovy/com/ullink/gradle/sonar/SonarScannerMsbuildPlugin.groovy (hunk 1 above)
Comment (apetres) [reply src/main/groovy/com/ullink/gradle/sonar/SonarScannerMsbuildPlugin.groovy:56]: In that command sonar-scanner-msbuild is ran which has the /? and /h arguments:
```
C:\sonar-scanner-msbuild\4.6.0.1930>SonarScanner.MSBuild.exe /h
SonarScanner for MSBuild 4.6
Using the .NET Framework version of the Scanner for MSBuild

Usage:

  SonarScanner.MSBuild.exe [begin|end] /key:project_key [/name:project_name] [/version:project_version] [/d:sonar.key=value] [/s:settings_file]

    When executing the begin phase, at least the project key must be defined.
    Other properties can dynamically be defined with '/d:'. For example, '/d:sonar.verbose=true'.
    A settings file can be used to define properties. If no settings file path is given, the file SonarQube.Analysis.xml in the installation directory will be used.
    Only the token should be passed during the end phase, if it was used during the begin phase.
```

File: src/main/groovy/com/ullink/gradle/sonar/SonarScannerMsbuildPlugin.groovy (hunk 2 above)
Comment (apetres) [reply src/main/groovy/com/ullink/gradle/sonar/SonarScannerMsbuildPlugin.groovy:28]: Yes, it exits with a non-zero exit code (thus failing the build) and with the same error message as the comment.

**Review by timotei — APPROVED** [review]

**Pull request merged**